framework/src/errors.ts

```diff
@@ -23,8 +24,20 @@ export class FrameworkError extends Error {
 	}
 }
 
+export class NonceNotReadyError extends FrameworkError {
+	public code = 'nonce_not_ready';
+	public actual: string;
+	public expected: string;
+	public constructor(message: string, actual: bigint, expected: bigint) {
+		super(message);
+		this.actual = actual.toString();
+		this.expected = expected.toString();
```

Comment:
Here again, you have to set the name of error explicitly for every custom error class. 

```bash
❯ node
Welcome to Node.js v12.18.3.
Type ".help" for more information.
> class MyError extends Error {}
undefined
> const e = new MyError("message")
undefined
> e.name
'Error'
```